core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/upscale/StackUpscaleActions.java

```diff
@@ -126,6 +130,9 @@ protected void doExecute(StackScalingFlowContext context, StackScaleTriggerEvent
                             hostGroupWithAdjustment.getValue(), context.isRepair());
                 }
 
+                if (!context.isRepair()) {
+                    stackService.save(targetedUpscaleSupportService.updateDnsResolverType(context.getStack()));
```

Comment:
- it's a bit odd for me to alter the stack in a `UPSCALE_PREVALIDATION_STATE`, I would expect validation steps here
- also it seems to me `updateDnsResolverType` calls `hostOrchestrator.unboundClusterConfigPresentOnAnyNodes` which is kind of a long running stuff, which should called in handlers instead of actions

Reply:
sure, I can move it to a handler, the point to execute it before any other salt call